### PR TITLE
docker-registry's middleware stanza should contain 'registry' and 'storage' by default

### PR DESCRIPTION
**Pull request by ganhuang**
Fix: https://bugzilla.redhat.com/show_bug.cgi?id=1356823

### CODE REVIEW

**Comment by ganhuang** [comment]
That's will be cool if bz#1365730 can be fixed in the same time so that our testing using s3 backend storage isn't blocked. :smile:

**Comment by sdodson** [comment]
LGTM, thanks

**Comment by sdodson** [comment]
aos-ci-test

File: roles/openshift_hosted/templates/registry_config.j2 [hunk 1]
```diff
@@ -58,6 +58,8 @@ auth:
   openshift:
     realm: openshift
 middleware:
+  registry:
+  - name: openshift
```
Comment (sdodson) [inline roles/openshift_hosted/templates/registry_config.j2]: After discussing with @miminar this config is only valid for 2.4.1 registry, so we need to make this conditional at least on 3.3 for now. Wrap this in 
```
{% if openshift.common.version_gte_3_3_or_1_3 | bool %}
{% endif -%}
```

File: roles/openshift_hosted/templates/registry_config.j2 [hunk 2]
```diff
@@ -69,4 +71,7 @@ middleware:
       baseurl: {{ openshift.hosted.registry.storage.s3.cloudfront.baseurl }}
       privatekey: {{ openshift.hosted.registry.storage.s3.cloudfront.privatekeyfile }}
       keypairid: {{ openshift.hosted.registry.storage.s3.cloudfront.keypairid }}
+{% else %}
+  storage:
+  - name: openshift
```
Comment (sdodson) [inline roles/openshift_hosted/templates/registry_config.j2]: Same here.

**Comment by sdodson** [comment]
aos-ci-test

**Comment by sdodson** [comment]
aos-ci-test

**Pull request merged**